modules/text_generation.py

```diff
@@ -314,24 +313,30 @@ def generate_with_streaming(**kwargs):
 def generate_reply_custom(question, original_question, seed, state, stopping_strings=None, is_chat=False):
     seed = set_manual_seed(state['seed'])
 
+    # Use 'encode' to truncate the prompt
+    truncated_prompt_ndarray = encode(question, truncation_length=get_max_prompt_length(state))
+
+    # Convert the truncated prompt back to a string (to avoid modifying 'generate' and 'generate_with_streaming')
+    truncated_prompt_string = shared.model.decode(truncated_prompt_ndarray[0])
+
```

Comment:
Use `encode` to truncate the prompt. Convert the prompt back to a string to be passed to `generate` and `generate_with_streaming`. I chose to do it this way because otherwise truncation logic would be duplicated in `LlamaCppModel`, `RWKVModel`, and `ExllamaModel`

Reply:
That makes sense. I had tried something similar (encode and then decode back) inside `LlamaCppModel` but never commited. Your solution is more general and elegant.